### ISSUE TITLE
Issues with running EvoMaster with JDK 17

### ISSUE THREAD

**Issue by jiaxian-lin** [EMResearch/EvoMaster#447]
```
javax.ws.rs.ProcessingException: java.lang.reflect.InaccessibleObjectException: Unable to make field protected java.lang.String java.net.HttpURLConnection.method accessible: module java.base does not "opens java.net" to unnamed module @7d9d1a19
        at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:309)
        at org.glassfish.jersey.client.JerseyInvocation.lambda$invoke$0(JerseyInvocation.java:630)
        at org.glassfish.jersey.client.JerseyInvocation.call(JerseyInvocation.java:665)
        at org.glassfish.jersey.client.JerseyInvocation.lambda$runInScope$3(JerseyInvocation.java:659)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
        at org.glassfish.jersey.internal.Errors.process(Errors.java:205)
        at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:390)
        at org.glassfish.jersey.client.JerseyInvocation.runInScope(JerseyInvocation.java:659)
        at org.glassfish.jersey.client.JerseyInvocation.invoke(JerseyInvocation.java:629)
        at org.evomaster.core.problem.rest.service.AbstractRestFitness.handleRestCall(AbstractRestFitness.kt:291)
        at org.evomaster.core.problem.rest.service.BlackBoxRestFitness.doCalculateCoverage(BlackBoxRestFitness.kt:60)
        at org.evomaster.core.problem.rest.service.BlackBoxRestFitness.doCalculateCoverage(BlackBoxRestFitness.kt:16)
        at org.evomaster.core.search.service.FitnessFunction.calculateIndividualWithPostHandling(FitnessFunction.kt:104)
        at org.evomaster.core.search.service.FitnessFunction.calculateCoverage(FitnessFunction.kt:56)
        at org.evomaster.core.search.service.FitnessFunction.calculateCoverage$default(FitnessFunction.kt:48)
        at org.evomaster.core.search.algorithms.RandomAlgorithm.searchOnce(RandomAlgorithm.kt:25)
        at org.evomaster.core.search.service.SearchAlgorithm.search(SearchAlgorithm.kt:66)
        at org.evomaster.core.Main$Companion.run(Main.kt:383)
        at org.evomaster.core.Main$Companion.initAndRun(Main.kt:147)
        at org.evomaster.core.Main$Companion.main(Main.kt:71)
        at org.evomaster.core.Main.main(Main.kt)
Caused by: java.lang.RuntimeException: java.lang.reflect.InaccessibleObjectException: Unable to make field protected java.lang.String java.net.HttpURLConnection.method accessible: module java.base does not "opens java.net" to unnamed module @7d9d1a19
        at org.glassfish.jersey.client.internal.HttpUrlConnector$3.run(HttpUrlConnector.java:518)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
        at org.glassfish.jersey.client.internal.HttpUrlConnector.setRequestMethodViaJreBugWorkaround(HttpUrlConnector.java:479)
        at org.glassfish.jersey.client.internal.HttpUrlConnector._apply(HttpUrlConnector.java:323)
        at org.glassfish.jersey.client.internal.HttpUrlConnector.apply(HttpUrlConnector.java:265)
        at org.glassfish.jersey.client.ClientRuntime.invoke(ClientRuntime.java:297)
        ... 21 common frames omitted
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make field protected java.lang.String java.net.HttpURLConnection.method accessible: module java.base does not "opens java.net" to unnamed module @7d9d1a19
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:354)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:297)
        at java.base/java.lang.reflect.Field.checkCanSetAccessible(Field.java:178)
        at java.base/java.lang.reflect.Field.setAccessible(Field.java:172)
        at org.glassfish.jersey.client.internal.HttpUrlConnector$3.run(HttpUrlConnector.java:513)
        ... 26 common frames omitted
```

**Comment by arcuri82** [comment]
Hi. 
 thx for reporting this issue. 
It is weird. Looks like some JDK problem. Can you provide the:

- whether you are using JAR file or an installer (Mac? Linux? Windows?)
- version of JDK you are using (if using JAR), and its name (can use `java --version`)
- version of EvoMaster
- command line options

**Comment by jiaxian-lin** [comment]
thanks for your replay,  I'm run in Windows, and here is my java version:
```
 java -version
java version "17.0.2" 2022-01-18 LTS
Java(TM) SE Runtime Environment (build 17.0.2+8-LTS-86)
Java HotSpot(TM) 64-Bit Server VM (build 17.0.2+8-LTS-86, mixed mode, sharing)
```

I use the latest EvoMaster release, and this is my command:
```
java -jar evomaster.jar --blackBox true --bbSwaggerUrl file://<api_file_path> --outputFormat JAVA_JUNIT_4 --maxTime 21600s --ratePerMinute 300 --bbTargetUrl 'domain'
```

**Comment by arcuri82** [comment]
Hi,
  thanks. This was a really interesting case... 
We do test with JDK 17, but it looks like there is a setting issue... and this issue leads to execute a `catch` in glassfish library that does not like JDK 17, and so the crash.

This is not something we can fix :( 
You have 3 workarounds:
- use `evomaster.exe` from the .msi installer (as it comes with its own JDK)
- "upgrade" your JDK to either version 8 or 11
-  try to add `--add-opens java.base/java.net=ALL-UNNAMED `  before  `-jar`, ie, `java --add-opens ... -jar evomaster.jar ...`

**Comment by jiaxian-lin** [comment]
thx!
evomaster.exe doesn't work too
I fix this error by adding --add-opens java.base/java.net=ALL-UNNAMED before -jar

**Comment by arcuri82** [comment]
thx.

`evomaster.exe` not working is indeed a bug, as it is wrongly packaged with JDK 17, which due to `--add-opens` madness should not really be used

**Comment by arcuri82** [comment]
This issue should now be fixed in the new 1.4.0 release
